src/admin/class-sign-in-with-google-admin.php

```diff
@@ -558,7 +558,7 @@ public function authenticate_user() {
 			$redirect = admin_url(); // Send users to the dashboard by default.
 		}
 
-		wp_redirect( apply_filters( 'login_redirect', $redirect ) ); //phpcs:ignore
+		wp_redirect( apply_filters( 'siwg_login_redirect', $redirect ) );
```

Comment:
I think it would be good to follow the same patterns as `login_redirect` ([docs](https://developer.wordpress.org/reference/hooks/login_redirect/)) and include the user in the args. This will allow somebody to redirect off of a particular user param if needed.

Reply:
On second thought... Why not just add the correct params for `login_redirect`?  That will make the plugin more widely compatible with other plugins without needing custom integrations. I am assuming that the use case for redirecting the user here is the exact same as when a user logs in like normal...

Reply:
I don't think that core filters are meant to be "re-defined" within plugins. Since `apply_filters()` [_invokes all functions attached to filter hook_](https://developer.wordpress.org/reference/functions/apply_filters/), adding the duplicate hook here would essentially double the number of functions being run right?

I feel like `apply_filters` should only be used with unique hooks to provide a single point of entry. That's my reasoning for changing this to a unique hook.

Reply:
I spent some time reviewing the source for [apply_filters()](https://github.com/WordPress/WordPress/blob/63ea8284a3e57ae289ce9ce39a4469cdc317906a/wp-includes/class-wp-hook.php#L280-L316) and I don't think it'll be an issue having a duplicate since it should only be called once. This method essentially bypasses [the login action](https://github.com/WordPress/WordPress/blob/5c2f80f60f01290cfdd885cd29855f6f5ea9e349/wp-login.php#L1244) so at any point, `login_redirect` should only be called once. Glad to have some discussion though!

Reply:
This is exactly what I was thinking. 👍